src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingActualsResponseMapper.java

```diff
@@ -191,27 +192,47 @@ private List<PlannedHearingDays> getPlannedHearingDays(HearingEntity hearingEnti
         return plannedHearingDays;
     }
 
-    private ArrayList<Party> setPartyDetails(HearingEntity hearingEntity) {
+    private ArrayList<Party> setPartyDetails(HearingEntity hearingEntity,
+                                             HearingDayDetailsEntity hearingDayDetailsEntity) {
 
         ArrayList<Party> partyDetailsList = new ArrayList<>();
-        for (HearingPartyEntity hearingPartyEntity : hearingEntity.getLatestCaseHearingRequest().getHearingParties()) {
+        for (HearingAttendeeDetailsEntity hearingAttendeeDetails
+            : hearingDayDetailsEntity.getHearingAttendeeDetails()) {
+
+            //Check if there is at least one match for party reference/id in hearing attendee and hearing party
+            HearingPartyEntity hearingPartyEntity = hearingEntity.getLatestCaseHearingRequest().getHearingParties()
+                .stream()
+                .filter(x -> x.getPartyReference().equals(hearingAttendeeDetails.getPartyId()))
+                .findFirst()
+                .orElse(null);
+
+            //Check if Case Hearing Id match in the Case Hearing Request and Hearing Party
+            boolean caseHearingIdMatches = hearingPartyEntity != null
+                && hearingEntity.getLatestCaseHearingRequest().getCaseHearingID()
+                    .equals(hearingPartyEntity.getCaseHearing().getCaseHearingID());
+
 
             Party partyDetails = new Party();
-            partyDetails.setPartyID(hearingPartyEntity.getPartyReference());
-            partyDetails.setPartyRole(hearingPartyEntity.getPartyRoleType());
-            //TBD separate ticket to be raised for subChannel
-            partyDetails.setPartyChannelSubType(null);
-            if (PartyType.IND.getLabel().equals(hearingPartyEntity.getPartyType().getLabel())) {
-                partyDetails.setIndividualDetails(setIndividualDetails(hearingPartyEntity));
+            partyDetails.setPartyID(hearingAttendeeDetails.getPartyId());
+            partyDetails.setPartyChannelSubType(hearingAttendeeDetails.getPartySubChannelType());
+            //Set PartyRole, IndividualDetails and OrganisationDetails
+            // if there is a match for CaseHearingId and party reference/id
+            if (hearingPartyEntity != null && caseHearingIdMatches) {
+                partyDetails.setPartyRole(hearingPartyEntity.getPartyRoleType());
+                if (PartyType.IND.getLabel().equals(hearingPartyEntity.getPartyType().getLabel())) {
+                    partyDetails.setIndividualDetails(setIndividualDetails(hearingPartyEntity));
+                } else {
+                    partyDetails.setOrganisationDetails(setOrganisationDetails(hearingPartyEntity));
+                }
+                //If there is no match just set the partyRole to null
             } else {
-                partyDetails.setOrganisationDetails(setOrganisationDetails(hearingPartyEntity));
+                partyDetails.setPartyRole(null);
```

Comment:
Do we need this here, wouldn't it be null at this point anyway ?

Reply:
done

Reply:
Updated in https://github.com/hmcts/hmc-cft-hearing-service/commit/f258ba42a8e3a5c96b9786cca080d3344e0f71c8